lib/project.js

```diff
@@ -0,0 +1,228 @@
+/**
+ * Copyright (c) 2017, Kinvey, Inc. All rights reserved.
+ *
+ * This software is licensed to you under the Kinvey terms of service located at
+ * http://www.kinvey.com/terms-of-use. By downloading, accessing and/or using this
+ * software, you hereby accept such terms of service  (and any agreement referenced
+ * therein) and agree that you have read, understand and agree to be bound by such
+ * terms of service and are of legal age to agree to such terms with Kinvey.
+ *
+ * This software contains valuable confidential and proprietary information of
+ * KINVEY, INC and is subject to applicable licensing agreements.
+ * Unauthorized reproduction, transmission or distribution of this file and its
+ * contents is a violation of applicable laws.
+ */
+
+const async = require('async');
+const chalk = require('chalk');
+const config = require('config');
+const KinveyError = require('./error.js');
+const logger = require('./logger.js');
+const prompt = require('./prompt');
+const util = require('./util.js');
+
+// todo convert es6 class > module.exports
+class Project {
+  constructor(path) {
+    this.app = null;
+    this.org = null;
+    this.service = null;
+    this.serviceName = null;
+    this.schemaVersion = null;
+    this.lastJobId = null;
+
+    this._selectService = (this._selectService).bind(this);
+    this._selectOrg = (this._selectOrg).bind(this);
+    this._selectAppOrOrg = (this._selectAppOrOrg).bind(this);
+    this._selectApp = (this._selectApp).bind(this);
+    this._execKinveyServices = (this._execKinveyServices).bind(this);
+    this.setup = (this.setup).bind(this);
```

Comment:
Are these binds necessary?  Since `setup`, `select`, etc are already class members, aren't they implicitly bound to the `this` of the class?